src/main/java/com/gooddata/http/client/LoginSSTRetrievalStrategy.java

```diff
@@ -94,7 +98,11 @@ public String obtainSst(final HttpClient httpClient, final HttpHost httpHost) th
             final HttpResponse response = httpClient.execute(httpHost, postLogin);
             int status = response.getStatusLine().getStatusCode();
             if (status != HttpStatus.SC_OK) {
-                throw new GoodDataAuthException("Unable to login: " + status);
+                Header requestIdHeader = response.getFirstHeader(X_GDC_REQUEST_HEADER_NAME);
+                HttpEntity responseEntity = response.getEntity();
+                String message = String.format("Unable to login reason='%s'. Request tracking details httpStatus=%s requestId=%s", (responseEntity != null) ? EntityUtils.toString(responseEntity) : null, status, (requestIdHeader != null) ? requestIdHeader.getValue() : null);
```

Comment:
Please extract both ternary operators to variables and wrap this line it it's too long.